### PR TITLE
Improve SDS detect version test to cover future releases.

### PR DESCRIPTION
**Pull request by ggbecker**
Taking advantage of `oscap info`, we can generate the datastream during runtime as it only needs to have the version set. This way we can extend the test to test future versions of SCAP we add in `schemas/sds`.

### CODE REVIEW

**Comment by ggbecker** [comment]
@openscap-ci test this please

File: tests/DS/sds_detect_version/test_detect_version.sh [hunk 1]
```diff
@@ -16,12 +16,21 @@ function test_oscap_info {
...
 	[ ! -s $stderr ]
 	grep -q "Version: $version" $stdout
 	rm $stdout
 	rm $stderr
+	rm $ds
 }
 
-test_oscap_info "1.2"
-test_oscap_info "1.3"
+SDS=$(find $builddir/schemas/sds -maxdepth 1 -mindepth 1 -type d -printf '%f\n')
```
Comment (jan-cerny) [inline tests/DS/sds_detect_version/test_detect_version.sh]: I think it should be `$top_srcdir` instead of `$builddir`.

Comment (ggbecker) [reply tests/DS/sds_detect_version/test_detect_version.sh]: I needed to add this variable to the `Makefile.am`, this was the only way to make it work. Is it correct?

Comment (jan-cerny) [reply tests/DS/sds_detect_version/test_detect_version.sh]: It seems the right way. But please don't rename it, keep `$top_srcdir` to be consistent across the whole project.

File: tests/DS/sds_detect_version/test_detect_version.sh [hunk 2]
```diff
@@ -16,12 +16,21 @@ function test_oscap_info {
 	version="$1"
 	stdout="$(mktemp)"
 	stderr="$(mktemp)"
-	$OSCAP info $srcdir/scap-$version-ds.xml > $stdout 2> $stderr
+	ds="$(mktemp)"
+	cp scap-ds.xml $ds
```
Comment (jan-cerny) [inline tests/DS/sds_detect_version/test_detect_version.sh]: missing `$srcdir`

File: tests/DS/sds_detect_version/test_detect_version.sh (hunk 1 above)
Comment (ggbecker) [reply tests/DS/sds_detect_version/test_detect_version.sh]: Done

File: tests/DS/sds_detect_version/test_detect_version.sh (hunk 2 above)
Comment (ggbecker) [reply tests/DS/sds_detect_version/test_detect_version.sh]: Done

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Comment by ggbecker** [comment]
rhel6 node failed in process58 test ;(

@openscap-ci test this please

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Comment by jan-cerny** [comment]
The CI fail is unrelated and has been fixed in https://github.com/OpenSCAP/openscap/pull/1344

**Pull request merged**